syntax = "proto3";

package fixbug;

option cc_generic_services = true;

message ResultCode{
    int32 errCode = 1;
    bytes errMsg = 2;
}

//登录请求
message LoginRequest{
    bytes name = 1;
    bytes pwd = 2;
}
//登录的响应
message LoginResponse{
    ResultCode result = 1;
    bool success = 2;
}

//注册的请求
message RegisterRequest{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}
//注册的响应
message RegisterResponse{
    ResultCode result = 1;
    bool success = 2;
}

//需要注册的服务
service UserServiceRpc{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}

